install/autodetect.go

```diff
@@ -41,13 +41,25 @@ func (p Parameters) checkDisabled(name string) bool {
 	return false
 }
 
-func (k *K8sInstaller) detectDatapathMode(helmValues map[string]interface{}) error {
+func (k *K8sInstaller) detectDatapathMode() error {
 	if k.params.DatapathMode != "" {
 		k.Log("ℹ️  Custom datapath mode: %s", k.params.DatapathMode)
 		return nil
 	}
 
-	routingMode, _, _ := unstructured.NestedString(helmValues, "routingMode")
+	vals, err := k.getHelmValues()
```

Comment:
The reason of the ci-ipsec-upgrade failure is that this `getHelmValues` call has the side effect of [configuring the affinities](https://github.com/cilium/cilium-cli/blob/0119a72bde163282a25d68ac0693c542e5071a13/install/helm.go#L114-L120) to not schedule Cilium pods on the selected nodes. However, `getHelmValues` is called explicitly during installation, while only indirectly here during upgrade. Dropping this call site caused affinities to be reverted on upgrade, in turn causing quite some failures. Let me fix that logic instead of reverting this commit.